app/experimenter/experiments/views.py

```diff
@@ -100,11 +143,75 @@ class ExperimentFilterset(filters.FilterSet):
         label="Show archived experiments", widget=forms.CheckboxInput()
     )
 
+    show_experiment_changes = filters.ChoiceFilter(
+        empty_label="No Date Restriction",
+        choices=[
+            ("starts_between", "Experiment Starts"),
+            ("pauses_between", "Experiment Pauses"),
+            ("ends_between", "Experiment Ends"),
+        ],
+        widget=forms.Select(attrs={"class": "form-control"}),
+        method="show_experiment_changes_filter",
+    )
+    start_date_range = filters.DateFromToRangeFilter(
+        method="date_range_filter",
+        widget=DateRangeWidget(
+            attrs={"type": "date", "class": "form-control range-field"}
+        ),
+    )
+    pause_date_range = filters.DateFromToRangeFilter(
+        method="date_range_filter",
+        widget=DateRangeWidget(
+            attrs={"type": "date", "class": "form-control range-field"}
+        ),
+    )
+    end_date_range = filters.DateFromToRangeFilter(
+        method="date_range_filter",
+        widget=DateRangeWidget(
+            attrs={"type": "date", "class": "form-control range-field"}
+        ),
+    )
+
     class Meta:
         model = Experiment
         form = ExperimentFiltersetForm
         fields = ExperimentFiltersetForm.Meta.fields
 
+    def show_experiment_changes_filter(self, queryset, name, value):
+        # this custom method isn't doing anything. There has to
+        # be a custom method to be able to display the select
+        # filter that controls which date range we show
+        return queryset
+
+    def date_range_filter(self, queryset, name, value):
+        experiments = queryset.all()
+        results = []
+        date = None
+
+        for experiment in experiments:
+            if name == "start_date_range":
+                date = experiment.start_date
+            elif name == "end_date_range":
+                date = experiment.end_date
+            elif name == "pause_date_range":
+                date = experiment.enrollment_end_date
```

Comment:
Yes, there was much talk/effort about doing this in the DB instead of loading the experiments into memory and then using the properties... and I definitely think it's theoretically possible... but it was proving pretty challenging.. I could definitely still hack away at that if we want me to.

Reply:
I don't know what Jared thinks -- personally, I don't think it's necessary at the scale that Experimenter works at, as long as there's a comment that expresses why we're doing it this way.

However, in thinking about this a little bit more, I think it would be fairly easy (as a minor optimization) do the "start date" comparison using the ORM, and just do the end date stuff in Python. What do you think?

Reply:
One other minor tweak here is that maybe it would be good to extract the mapping of filter name -> attribute name outside the loop. Something maybe like this:

```
attributes = {
    "start_date_range": "start_date",
    "end_date_range": "end_date",
    "pause_date_range": "enrollment_end_date"
}
attribute = attribute_names[name]

for experiment in experiments:
    date = getattr(experiment, attribute)
    # ...
```

Reply:
I still think it would be good to move the logic about which field to look at outside the loop, since it doesn't change on different iterations of the loop.

Reply:
I definitely thought it over but it's hard for me to see how this is an improvement.. do you think it's a better performance thing?

Reply:
I think the performance difference is likely to be negligible. Mostly I just think it feels cleaner to put stuff in the loop only when it needs to rerun on each loop iteration. The way it's written right now slightly implies that the loop could change `name`. But I don't feel strongly about it, so I'll just mark this as resolved.